app/controllers/services/lifecycle/service_key_manager.rb

```diff
@@ -0,0 +1,59 @@
+module VCAP::CloudController
+  class ServiceKeyManager
+    class ServiceInstanceNotFound < StandardError; end
+    class ServiceInstanceNotBindable < StandardError; end
+
+    def initialize(services_event_repository, access_validator, logger)
+      @services_event_repository = services_event_repository
+      @access_validator = access_validator
+      @logger = logger
+    end
+
+    def create_service_key(request_attrs)
+      service_instance = ServiceInstance.first(guid: request_attrs['service_instance_guid'])
+      raise ServiceInstanceNotFound unless service_instance
```

Comment:
Ok, I suppose that's ok for now. 

For future reference, please don't add code if you aren't going to write a corresponding test. For example, I was originally planning to ask for removal of the following line:
`safe_unbind_instance(service_key)`
because it has no tests (because they are part of the orphan mitigation suite that you're planning to add later). We didn't immediately understand why that line was there, and assumed it was part of this story. It's good that you're thinking ahead, but it is easier for us to follow which logic pertains to the current story if it's kept as minimal as possible. Also, we'll have to remember that this particular line of code is untested and double-check test coverage when the orphan mitigation story is submitted. This separation of tests and code will create more work for us in the future.

Rather than ask you to remove code and then re-add it later, we can keep it so long as the subsequent story about orphan mitigation includes backfilled tests. I've added a task in the story.